docs/fiddles/features/progress-bar/main.js

```diff
@@ -1,21 +1,45 @@
 const { app, BrowserWindow } = require('electron')
 
+let indeterminateTimeout, progressInterval
+
 function createWindow () {
   const win = new BrowserWindow({
     width: 800,
-    height: 600,
-    webPreferences: {
-      nodeIntegration: true
-    }
+    height: 600
   })
 
   win.loadFile('index.html')
-  win.setProgressBar(0.5)
-}
 
+  const INCREMENT = 0.02
+  const INTERVAL_DELAY = 100 // ms
+  const INDETERMINATE_PAUSE = 5000 // ms
+
+  let c = 0
+  progressInterval = setInterval(() => {
+    // update progress bar to next value
+    win.setProgressBar(c)
+
+    if (c > 1) {
+      // set to indeterminate state
+      win.setProgressBar(2)
+
+      // stop the interval
+      clearInterval(progressInterval)
+
+      // and after INDETERMINATE_PAUSE reset the progress bar
+      indeterminateTimeout = setTimeout(() => win.setProgressBar(-1), INDETERMINATE_PAUSE)
```

Comment:
I was thinking it would be awesome if it just kept looping (disappeared for a few seconds, then started over at 0, filled up, etc)

Not a big deal though.

Reply:
I went ahead and did this. It ended up simplifying the code a bit because it got rid of the second timer I had. It was enough of a change that I would appreciate another look though.